### PR TITLE
Razor 16.8 Preview 4 Release

### PR DESCRIPTION
**Pull request by TanayParikh**
* Razor formatting improvements ([dotnet/aspnetcore-tooling#25470](https://github.com/dotnet/aspnetcore/issues/25470)) / ([dotnet/aspnetcore-tooling#14271](https://github.com/dotnet/aspnetcore/issues/14271))
  * Add using for C# Type light bulb ([dotnet/aspnetcore-tooling#18173](https://github.com/dotnet/aspnetcore/issues/18173))
  * Fully qualify C# Type light bulb ([dotnet/aspnetcore-tooling#24778](https://github.com/dotnet/aspnetcore/issues/24778))

cc/ @NTaylorMullen @ajaybhargavb @ryanbrandenburg

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3155,6 +3155,54 @@
               "env": {
                 "type": "object",
                 "description": "Environment variables passed to dotnet. Only valid for hosted apps."
+              },
+              "logging": {
```
Comment (TanayParikh) [inline package.json:3159]: fyi/ @captainsafia

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,7 +15,9 @@
 * Updated Razor support
   * Updated OmniSharp version (should improve stability) [dotnet/aspnetcore-tooling#20320](https://github.com/dotnet/aspnetcore/issues/20320)
   * Corrected positioning for `@using` for components added by light bulb. [dotnet/aspnetcore-tooling#25019](https://github.com/dotnet/aspnetcore/issues/25019)
-  * Razor formatting improvements
+  * Razor formatting improvements ([dotnet/aspnetcore-tooling#25470](https://github.com/dotnet/aspnetcore/issues/25470)) / ([dotnet/aspnetcore-tooling#14271](https://github.com/dotnet/aspnetcore/issues/14271))
+  * Add using for C# Type light bulb ([dotnet/aspnetcore-tooling#18173](https://github.com/dotnet/aspnetcore/issues/18173))
+  * Fully qualify C# Type light bulb ([dotnet/aspnetcore-tooling#24778](https://github.com/dotnet/aspnetcore/issues/24778))
```
Comment (captainsafia) [inline CHANGELOG.md:20]: - Added support for engine logging on .NET process for Blazor WASM apps ([OmniSharp/omnisharp-vscode#4070](https://github.com/OmniSharp/omnisharp-vscode/issues/4070))
- Fixed bug in clean-up of Blazor WASM debugging session ([OmniSharp/omnisharp-vscode#4056](https://github.com/OmniSharp/omnisharp-vscode/issues/4056))

Comment (TanayParikh) [reply CHANGELOG.md:20]:
```suggestion
  * Fully qualify C# Type light bulb ([dotnet/aspnetcore-tooling#24778](https://github.com/dotnet/aspnetcore/issues/24778))
  * Added support for engine logging on .NET process for Blazor WASM apps (OmniSharp/omnisharp-vscode#4070)
  * Fixed bug in clean-up of Blazor WASM debugging session (OmniSharp/omnisharp-vscode#4056)

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,7 +15,12 @@
...
+  * Added support for engine logging on .NET process for Blazor WASM apps (OmniSharp/omnisharp-vscode#4070)
+  * Fixed bug in clean-up of Blazor WASM debugging session (OmniSharp/omnisharp-vscode#4056)
```
Comment (TanayParikh) [inline CHANGELOG.md:22]:
```suggestion
  * Added support for engine logging on .NET process for Blazor WASM apps ([OmniSharp/omnisharp-vscode#4070](https://github.com/OmniSharp/omnisharp-vscode/issues/4070))
  * Fixed bug in clean-up of Blazor WASM debugging session ([OmniSharp/omnisharp-vscode#4056](https://github.com/OmniSharp/omnisharp-vscode/issues/4056))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,7 +15,12 @@
 * Updated Razor support
   * Updated OmniSharp version (should improve stability) [dotnet/aspnetcore-tooling#20320](https://github.com/dotnet/aspnetcore/issues/20320)
   * Corrected positioning for `@using` for components added by light bulb. [dotnet/aspnetcore-tooling#25019](https://github.com/dotnet/aspnetcore/issues/25019)
-  * Razor formatting improvements
+  * Razor formatting improvements ([dotnet/aspnetcore-tooling#25470](https://github.com/dotnet/aspnetcore/issues/25470)) / ([dotnet/aspnetcore-tooling#14271](https://github.com/dotnet/aspnetcore/issues/14271))
```
Comment (NTaylorMullen) [inline CHANGELOG.md:18]: Might want to make this more pronounced to stand out for users reading the release notes.
```suggestion
  * Mixed HTML & C# Razor formatting support ([dotnet/aspnetcore-tooling#25470](https://github.com/dotnet/aspnetcore/issues/25470)) / ([dotnet/aspnetcore-tooling#14271](https://github.com/dotnet/aspnetcore/issues/14271))
```

**Review by NTaylorMullen — APPROVED** [review]

**Comment by TanayParikh** [comment]
cc/ @JoeRobich

**Review by JoeRobich — APPROVED** [review]

**Comment by TanayParikh** [comment]
Thanks @JoeRobich! 🎉

**Comment by JoeRobich** [comment]
@TanayParikh In case you didn't find it, I built a new prerelease with this change - https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.23.3-beta3

**Comment by TanayParikh** [comment]
> @TanayParikh In case you didn't find it, I built a new prerelease with this change - https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.23.3-beta3


Thanks!

**Pull request merged**